core/CronArchive.php

```diff
@@ -1742,20 +1738,19 @@ private function getUrlsWithSegment($idSite, $period, $date)
             $urlWithSegment = $this->getVisitsRequestUrl($idSite, $period, $dateParamForSegment, $segment);
             $urlWithSegment = $this->makeRequestUrl($urlWithSegment);
 
-            $periodInProgress = $this->isAlreadyArchivingAnyLowerOrThisPeriod($idSite, $period);
-            if ($periodInProgress) {
-                $this->logger->info("- skipping segment archiving for period '{period}' with segment '{segment}' because processing the period '{periodcheck}' is already in progress.", array('segment' => $segment, 'period' => $period, 'periodcheck' => $periodInProgress));
-                continue;
-            }
-
-            if ($cliMulti->isCommandAlreadyRunning($urlWithSegment)) {
-                $this->logger->info("- skipping segment archiving for '{segment}' because such a process is already in progress.", array('segment' => $segment));
+            if ($this->isAlreadyArchivingSegment($urlWithSegment, $idSite, $period, $segment)) {
                 continue;
             }
 
             $request = new Request($urlWithSegment);
             $logger = $this->logger;
-            $request->before(function () use ($logger, $segment, $segmentCount, &$processedSegmentCount) {
+            $self = $this;
+            $request->before(function () use ($logger, $segment, $segmentCount, &$processedSegmentCount, $idSite, $period, $urlWithSegment, $self) {
+
+                if ($self->isAlreadyArchivingSegment($urlWithSegment, $idSite, $period, $segment)) {
+                    return Request::ABORT;
+                }
+
                 $processedSegmentCount++;
```

Comment:
actually wondering if I should increase `$processedSegmentCount` before the `if` even when aborted? @mattab ?

Reply:
I'm not sure if it changes anything as it seems only used for logging info on screen. Since it's a very edge case, it does not really matter. Would probably leave it as it is?